### DISCUSSION TITLE
check effectiveness of initial guess

### DISCUSSION THREAD

**Discussion by LoreMoretti** [casadi/casadi#3411] [Q&A]
Hello,

I am solving a constrained NLP leveraging the Opti stack. I am using `ipopt` and `ma97`.

I am providing an initial guess for the optimization variables to warm-start the solver.

If I look at the `ipopt` iterations, I see that the initial cost is in the order of `1e4`. Then, after few iterations, I see the cost increases up to the order of `1e7` and it takes a while to go back to values close to the initial one.

From my experience, the initial guess I am providing should be a good one.

Is there anything I can check to understand what is going on?

Thanks a lot for your help,

Lorenzo

**Comment by brunoml5** [comment]
Hi!

First, did you set the `ipopt` option `warm_start_init_point` to `yes`? See if this make a difference. Also, take a look at this [part of the docs](https://web.casadi.org/docs/#problem-solving-and-retrieving) so that you are providing the dual variables properly.

I have a suggestion, maybe it helps. First, solve an easier NLP problem providing good initial guesses for the optimization variables, maybe remove the objective function and leave the constraints. The result is primal values (`x`) and the dual values (`lam_g`). Then, include the objective function and run the solver again, see if it helps.

Nevertheless, `ipopt` is based on the interior point method and this class of solvers does not exploit warm-starting very well. Maybe switch to an `SQP` method.

I hope it helps!

**Comment by sandeep026** [comment]
Hope this helps .(https://github.com/casadi/casadi/wiki/FAQ%3A-Warmstarting-with-IPOPT)
